specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/iotcentral.json

```diff
@@ -478,8 +575,7 @@
     },
     "DeviceCloudProperties": {
       "description": "Cloud property values associated with the device.",
-      "type": "object",
-      "additionalProperties": true
+      "type": "object"
```

Comment:
This means this is an empty object, and nothing exists inside, which is surprising?

Reply:
That doesn't match the example usage:
https://github.com/Azure/azure-rest-api-specs/blob/33aa6e55353e7c4ae98aafe8be2890be3ae59c56/specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2019-10-28-preview/examples/devices_cloudproperties_get.json

Where there is a AssetID.

Could you detail what you want here?

Reply:
You're correct about the intended behavior, though according to JSON Schema Draft 4 a `true` value and an absent one are synonymous (restricting to an empty object would require an explicit `false`). We've had trouble with AutoRest and `"additionalProperties": true`, which is why this was removed.

Reply:
Unless you have trouble with autorest, I would stick with an addProp object:

https://github.com/Azure/azure-rest-api-specs/blob/6879889f69c84fb26a196121eab5d79c19e87ffb/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json#L4231-L4237

Reply:
I am uncertain what we were seeing before, but `additionalProperties: true` appears to be working correctly again, so we can revert this.

Reply:
added it back in